### PR TITLE
Fixing setting up custom shell in container terminal

### PR DESCRIPTION
**Pull request by VojtechBartos**
Hey guys,
i wanted to set custom shell to `bash` instead of `sh` which is default. I set environment `ENV SHELL /bin/bash`, but it was still picking up default option `sh` every time. I have found in code when you trying to check if somebody wants to set custom shell that you are accessing to array like to object, so i have added simple reduce function to make it actually object.

Hopefully everything is ok ;)

Let me know if you have any questions.
Vojta

Signed-off-by: Vojta Bartos <hi@vojtech.me>

### CODE REVIEW

File: src/components/ContainerDetailsSubheader.react.js [hunk 1]
```diff
@@ -102,7 +102,11 @@ var ContainerDetailsSubheader = React.createClass({
     if (!this.disableTerminal()) {
       metrics.track('Terminaled Into Container');
       var container = this.props.container;
-      var shell = ContainerUtil.env(container).SHELL;
+      var shell = ContainerUtil.env(container).reduce((envs, env) => {
+        envs[env[0]] = env[1];
+        return envs;
+      }, {}).SHELL;
+
       if(typeof shell === 'undefined') {
```
Comment (jeffdm) [inline src/components/ContainerDetailsSubheader.react.js]: Would it be better to just do `if (!shell)` ?

Comment (VojtechBartos) [inline src/components/ContainerDetailsSubheader.react.js]: @jeffdm yeah, definitely, but i wanted to do just necessary steps to make it work. Anyway, i have changed it.

**Comment by jeffdm** [comment]
Thanks for the PR! Code LGTM

@mchiang0610 can you test functionality

**Comment by mchiang0610** [comment]
@jeffdm anything specific to run it with?

**Comment by jeffdm** [comment]
Not much to test-  containers with $SHELL = `bash` should start with bash now instead of `sh`.

Merging

**Pull request merged**